src/MICore/Transports/PipeTransport.cs

```diff
@@ -24,13 +24,38 @@ public class PipeTransport : StreamTransport
         private bool _killOnClose;
         private bool _filterStderr;
         private int _debuggerPid = -1;
+        private string _pipePath;
+        private string _cmdArgs;
 
         public PipeTransport(bool killOnClose = false, bool filterStderr = false, bool filterStdout = false) : base(filterStdout)
         {
             _killOnClose = killOnClose;
             _filterStderr = filterStderr;
         }
 
+        public bool Interrupt(int pid)
+        {
+            if (_cmdArgs == null)
+            {
+                return false;
+            }
+
+            Process proc = new Process();
+            string killCmd = string.Format(CultureInfo.InvariantCulture, "kill -2 {0}", pid);
+            proc.StartInfo.FileName = _pipePath;
+            proc.StartInfo.Arguments = string.Format(CultureInfo.InvariantCulture, _cmdArgs, killCmd);
+            proc.StartInfo.WorkingDirectory = System.IO.Path.GetDirectoryName(_pipePath);
+            proc.EnableRaisingEvents = false;
+            proc.StartInfo.RedirectStandardInput = false;
+            proc.StartInfo.RedirectStandardOutput = false;
+            proc.StartInfo.RedirectStandardError = false;
+            proc.StartInfo.UseShellExecute = false;
+            proc.StartInfo.CreateNoWindow = true;
+            proc.Start();
+            proc.WaitForExit();
+            return true;
```

Comment:
If things fail it is going to be really tricky to figure out why. Should we at least grab stderr/out and dump it to the log? Should we check to see if the pipe command succeeds?